### PR TITLE
Refactor: Stop adding arbitrary attributes to module obj when building

### PR DESCRIPTION
**Pull request by thejcannon**
## Steps

- [ ] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [ ] Write a good description on what the PR does.

## Description

Moving code from `_post_build` into `_data_build` so that we don't have to attach arbitrary objects to the module (namely `_import_from_nodes`, and `_delayed_assattr`).

Alternatively we could:
- Keep the code in `_post_build` and pass the info along
- Same, but pass the builder

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
|    | :bug: Bug fix          |
|    | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |
|   | :scroll: Docs          |

### CODE REVIEW

File: astroid/builder.py [hunk 1]
```diff
@@ -190,8 +179,18 @@ def _data_build(self, data, modname, path):
             )
         builder = rebuilder.TreeRebuilder(self._manager, parser_module)
         module = builder.visit_module(node, modname, node_file, package)
-        module._import_from_nodes = builder._import_from_nodes
-        module._delayed_assattr = builder._delayed_assattr
+
+        self._manager.cache_module(module)
```
Comment (DanielNoord) [inline astroid/builder.py:183]: Shouldn't the `cache_module` stay in `_post_build`? If we ever do something different in `_data_build` after this line or we add a method between `_data_build` and `_post_build` it won't be cached?

**Review by DanielNoord — COMMENTED** [review]
Based on the tests and the actual code I this change seems fine? It doesn't actually do much different, other than making the code more readable since `_post_build` is called after `_data_build` anyway.

Edit: And by the way @thejcannon sorry for letting this PR drop down into "stale-PR-hell" 😅

File: astroid/builder.py (hunk 1 above)
Comment (cdce8p) [reply astroid/builder.py:183]: I think it wouldn't matter. The `module` object reference is cached, ie. if the object changes, those changes are also reflected in the cache.

**Review by cdce8p — DISMISSED** [review]
I would prefer if we kept the logic inside `_post_build`. IMO passing the `builder` along is probably the better alternative.

**Comment by cdce8p** [comment]
I've updated the code to pass `builder` along instead. Didn't want to drag this on any longer.

**Review by cdce8p — APPROVED** [review]

**Review by Pierre-Sassoulas — APPROVED** [review]
LGTM thank you @thejcannon and @cdce8p

**Pull request merged**